src/components/Tutorial/__tests__/Tutorial.ts

```diff
@@ -0,0 +1,260 @@
+      await user.keyboard('{Enter}')
+      await act(vi.runOnlyPendingTimersAsync)
+
+      expect(screen.getByText('Well done!')).toBeInTheDocument()
+    })
+  })
+
+  it('we create a second thought', async () => {
+    expect(screen.getByText(/Try adding another thought/)).toBeInTheDocument()
+    await user.type(lastEmptyThought(), 'second')
+    await user.keyboard('{Enter}')
+    await act(vi.runOnlyPendingTimersAsync)
+
+    expect(screen.getByText('Now type some text for the new thought.')).toBeInTheDocument()
+  })
+
+  it('we create a third thought', async () => {
+    await user.type(lastEmptyThought(), 'third')
+    await user.keyboard('{Enter}')
+    await act(vi.runOnlyPendingTimersAsync)
+
+    expect(screen.getByText(/Hit the Delete key to delete the current blank thought/)).toBeInTheDocument()
+  })
+
+  it('we learn to create nested thoughts', async () => {
+    /* thoughts:
+  - first
+  - second
+  - third
+  -
+    */
+    await user.keyboard('{Backspace}')
+    await user.keyboard('{Control>}{Enter}{/Control}')
+    await act(vi.runOnlyPendingTimersAsync)
+
+    await user.type(lastEmptyThought(), 'child of third')
+    await act(vi.runOnlyPendingTimersAsync)
+
+    expect(screen.getByText(/As you can see, the new thought "child of third" is nested/)).toBeInTheDocument()
+  })
+
+  describe('step autoexpand', async () => {
+    it('we learn about automatic thought hiding', async () => {
+      await user.click(screen.getByText('Next'))
+      expect(screen.getByText(/thoughts are automatically hidden when you click away/)).toBeInTheDocument()
+    })
+
+    it('we hide nested thoughts by clicking away', async () => {
+      /* thoughts:
+      - first
+      - second
+      - third
+        - child of third
+      */
+      await user.click(screen.getByText('second'))
+      expect(screen.getByText(/Notice that "child of third" is hidden now/)).toBeInTheDocument()
+    })
+
+    it('we reveal hidden thoughts by clicking parent', async () => {
+      await user.click(screen.getAllByText('third').at(-1)!)
+      expect(screen.getByText('child of third')).toBeInTheDocument()
+    })
+  })
+
+  describe('step success', async () => {
+    it('we complete the first tutorial', async () => {
+      expect(screen.getByText(/Lovely\. You have completed the tutorial/)).toBeInTheDocument()
+    })
+
+    it('we can exit tutorial mode', async () => {
+      await user.click(screen.getByText('Play on my own'))
+      await act(vi.runOnlyPendingTimersAsync)
+      expect(() => screen.getByTestId('tutorial-step')).toThrow('Unable to find an element')
+    })
+
+    it('we can continue to advanced tutorial', async () => {
+      expect(screen.getByText('Learn more')).toBeInTheDocument()
+    })
+  })
+})
+
+describe('Tutorial 2', async () => {
+  beforeEach(() => createTestApp({ tutorial: true }))
+  afterEach(cleanupTestEventHandlers)
+  afterAll(cleanupTestApp)
+  it('we learn about context indicators', async () => {
```

Comment:
It's more accurate to say "contexts" or "context view" than "context indicators". Let's update this to be more consistent with **em** terminology.

Likewise below.